### PR TITLE
Fix favorite faction display

### PR DESCRIPTION
**Pull request by Syldarion**
These changes include a fix to the favorite faction display and an update to the formatting of the changed file. I'll remove the latter if it isn't wanted, it was just to make working on it easier.

The favorite faction fix is explained in the second commit message.

### CODE REVIEW

**Review by Brutus5000 — CHANGES_REQUESTED** [review]
Generally I don't have a problem with your code, but given the fact that we recently spent multiple hours debugging performance issues of this application I cannot accept 3 nested for loops inside a while loop.

We need to find better data structures for this.

**Comment by Syldarion** [comment]
For sure, I'll take another stab at this then.

**Comment by norraxx** [comment]
@Syldarion Maybe, we could talk about this code at discord. You would explain what you want to achieve, and I'd say what I don't like on this commit, what do you think?

**Comment by Syldarion** [comment]
@norraxx Basically, just trying to have favorite faction percentage based off all games rather than those found first. I can see how just re-using a nested for loop is undesirable, seeing as it'll be going through the same data and processing it the same way twice.

**Comment by norraxx** [comment]
@Syldarion Imagine how much load you have to do, to just compute that stuff. As I said, let's talk about that problem on discord https://discord.gg/VYX25M.
Write me there, i'm `dragonite` user there, maybe we can figure how to make it better. As @Brutus5000 said your solution will make very big load on servers, web page will start minutes, and maybe you won't have enough memory, and what you want is compute 4 numbers. :-)

File: scripts/updateLeagues.js [hunk 1]
```diff
@@ -5,244 +5,260 @@ let fs = require('fs');
...
+                        'winner': null,
+                        'participants': [],
+                        'startTimestamp': moment(record.attributes.startTime).unix(),
+                        'endTimestamp': moment(record.attributes.endTime).unix()
                     }
                 }
-                
+
+                if (leagueData.games == undefined) leagueData.games = [];
+                if (leagueData.playerData == undefined) leagueData.playerData = [];
```
Comment (norraxx) [inline scripts/updateLeagues.js:63]: I don't like conditions like that, use brackets, it's ok to have 3 lines instead of 1.

**Comment by Syldarion** [comment]
@norraxx I spent some time trying to come up with a better solution, but only managed to find either longer runtime or wrong data.
I'd ask that you try the change yourself. Locally, running with this change is within a couple seconds of the runtime of what is live, but comes with accurate data.
Or I could close this and open an issue.

**Comment by Syldarion** [comment]
As mentioned in the commit message of what I just added, an obvious solution had been staring me in the face all along. The for loop I added previously is gone, and performance seems in line with live.

**Comment by norraxx** [comment]
I would to meet on discord.

File: scripts/updateLeagues.js [hunk 2]
```diff
@@ -5,244 +5,251 @@ let fs = require('fs');
...
+                let categories = {};
+
+                for (let k in games) {
+                    if (leagueData.games.indexOf(k) > -1) {
+                        continue;   // Game has already been calculated, skipping it
+                    }
+                    const game = games[k];
+                    const timeElapsed = game.endTimestamp - game.startTimestamp;
+
+                    for (l in game.participants) {
```
Comment (norraxx) [inline scripts/updateLeagues.js:108]: use let

File: scripts/updateLeagues.js [hunk 3]
```diff
@@ -5,244 +5,251 @@ let fs = require('fs');
...
+                            'rating': playerRecord.rating,
+                            'points': 0,
+                            'factions': playerRecord.factions,
+                            'wld': { "w": 0, "l": 0, "d": 0 },
+                            'secondsPlayed': timeElapsed
+                        };
+
+                        // Let's check if we don't have that player already
+                        let already = false;
+                        for (category in leagueData.playerData) {
```
Comment (norraxx) [inline scripts/updateLeagues.js:123]: -//-

File: scripts/updateLeagues.js [hunk 4]
```diff
@@ -5,244 +5,251 @@ let fs = require('fs');
...
-                            if (!already){
-                                const catName = getRangeName(leaguesConfig.ratingRange, player.rating);
-                                let found = false;
-                                for (categoryIndex in leagueData.playerData){
-                                    if (leagueData.playerData[categoryIndex].categoryName == catName){
-                                        leagueData.playerData[categoryIndex].data.push(player);
-                                        found = true;
+                }
+
+                let categories = {};
```
Comment (norraxx) [inline scripts/updateLeagues.js:99]: you don't need it

**Review by norraxx — COMMENTED** [review]
Cosmetic changes.

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**